.devcontainer/configure.py

```diff
@@ -0,0 +1,222 @@
+        "https://docs.docker.com/compose/") from e
+
+  docker_image_key = "base"
+
+  # ------------------------------------------------------------------------- #
+  #                               Mandatory Steps                             #
+  # ------------------------------------------------------------------------- #
+
+  # STEP 1: Verify the user doesn't have a pre-existing `docker-compose.yml`.
+  #         If yes, confirm they want to overwrite it.
+  if os.path.isfile(CURRENT_DIR / "docker-compose.yml"):
+    if get_input(
+        "A `docker-compose.yml` already exists. Are you certain you want to overwrite it [y/N]?",
+        default_answer="n",
+        accepted_answers=["y", "n"]) == "n":
+      sys.exit(1)
+
+  # STEP 2: Read the template configuration file
+  with open(CURRENT_DIR / "docker-compose.base.yml") as f:
+    docker_config = json.load(f)
+
+  # STEP 3: Prebuilt vs Locally Built Containers
+  use_official_img = get_input(
+      "Do you wish to use the official prebuild development containers [Y/n]?",
+      default_answer="y",
+      accepted_answers=["y", "n"]) == "y"
+
+  # ------------------------------------------------------------------------- #
+  #                 Optional Mounting Points - Build & Cache                  #
+  # ------------------------------------------------------------------------- #
+
+  # STEP 4 [OPTIONAL]: Does the user want to mount a directory for CCACHE ?
+  ccache_mount_cache = get_input_path(
+      "[OPTIONAL] Enter the the directory path to mount as CCache Cache.\n"
+      "  It will be mounted in the container under: `/home/iree/.cache/ccache` [Default: None]:"
+  )
+  if ccache_mount_cache:
+    docker_config["services"]["iree-dev"]["volumes"].append(
+        f"{ccache_mount_cache}:/home/iree/.cache/ccache:cached")
+
+  # STEP 5 [OPTIONAL]: Does the user want to mount a directory for BUILD ?
+  build_mount_cache = get_input_path(
+      "[OPTIONAL] Enter the the directory path to mount as Build Directory.\n"
+      "  It will mounted in the container under: `/home/iree/build` [Default: None]:"
+  )
+  if build_mount_cache:
+    docker_config["services"]["iree-dev"]["volumes"].append(
+        f"{build_mount_cache}:/home/iree/build:cached")
+
+  # ------------------------------------------------------------------------- #
+  #                Optional Deep Learning Accelerator Support                 #
+  # ------------------------------------------------------------------------- #
+
+  # STEP 6 [OPTIONAL]: Does the user want to use NVIDIA GPUs ?
+  use_nvidia_gpu = False
+  nvidia_gpu_available, err_msg = is_nvidia_gpu_available()
+
+  if nvidia_gpu_available:
+    use_nvidia_gpu = get_input(
+        "[OPTIONAL] Do you wish to use NVIDIA GPU [y/N]?",
+        default_answer="n",
+        accepted_answers=["y", "n"]) == "y"
+
+    if use_nvidia_gpu:
+      docker_image_key = "nvidia"
+    else:
+      del docker_config["services"]["iree-dev"]["deploy"]
```

Comment:
This only deletes the deploy option if an NVIDIA GPU is available. If an NVIDIA GPU isn't available you still have it!

Reply:
Oh right ! Sorry

Reply:
Fixed